query/logicalplan/optimize.go

```diff
@@ -8,12 +8,18 @@ type Optimizer interface {
 	Optimize(plan *LogicalPlan) *LogicalPlan
 }
 
-var DefaultOptimizers = []Optimizer{
-	&AverageAggregationPushDown{},
-	&PhysicalProjectionPushDown{},
-	&FilterPushDown{},
-	&DistinctPushDown{},
-	&ProjectionPushDown{},
+func DefaultOptimizers() []Optimizer {
+	return []Optimizer{
+		&AverageAggregationPushDown{},
+		&PhysicalProjectionPushDown{
+			defaultProjections: []Expr{
+				RegExpNotColumnMatch("^hashed."),
```

Comment:
Is the cost of running this extra projection what you measured?

Reply:
The measurements are prior to this change and then against this change with prehashed columns.

For schemas without prehashed columns this projection shouldn't result in much of anything

Reply:
```
name                  old time/op    new time/op    delta
Aggregation/sum-10       406µs ± 1%     403µs ± 1%  -0.53%  (p=0.008 n=10+9)
Aggregation/count-10     400µs ± 1%     397µs ± 0%  -0.59%  (p=0.000 n=9+10)
Aggregation/max-10       409µs ± 0%     407µs ± 1%  -0.45%  (p=0.001 n=10+10)

name                  old alloc/op   new alloc/op   delta
Aggregation/sum-10       426kB ± 0%     426kB ± 0%    ~     (p=0.190 n=10+10)
Aggregation/count-10     424kB ± 0%     425kB ± 0%    ~     (p=0.536 n=10+7)
Aggregation/max-10       426kB ± 0%     426kB ± 0%    ~     (p=0.122 n=8+10)

name                  old allocs/op  new allocs/op  delta
Aggregation/sum-10         788 ± 0%       788 ± 0%    ~     (all equal)
Aggregation/count-10       787 ± 0%       787 ± 0%    ~     (all equal)
Aggregation/max-10         788 ± 0%       788 ± 0%    ~     (all equal)
```

Here's the benchmark run before and after this change without modifications to the benchmark

Reply:
Do we have measurements of a vanilla selection with and without this projection? I agree with you though, I don't think it should be that high overhead given it's per-record

Reply:
I don't have one no